coprocess.go

```diff
@@ -111,6 +111,12 @@ func (c *CoProcessor) ObjectFromRequest(r *http.Request) *coprocess.Object {
 			"OrgID": c.Middleware.Spec.OrgID,
 			"APIID": c.Middleware.Spec.APIID,
 		}
+
+		//Append CoProccess extractors
+		for k, v := range c.Middleware.Spec.CoProccessSpecExtras {
+			object.Spec[k] = v
+		}
```

Comment:
This probably needs nesting in a sub-object, such as `ConfigData`. Otherwise, if you put `"APIID": "foo"` in the API definition, you'll overwrite the real APIID field.

Reply:
Hi, sure I can elaborate on the issues with protobuf3.

1) The issue is that protobuf3 doesn't have a value that directly translates to the go object of `map[string]interface{}`.  The closest that we can generate is using the `Any` value in protobuf.  Unfortunately, this doesn't create the required object in Go, but rather a `*Any` which will require using reflection and an intermediate type to assign values to it.  Additionally, even if we did an intermediate to resolve the field, we would need to implement the same for the other languages that the protobuf is responsible for.  It's a much easier solution to conform to a map[string]string since that translates fine in all the languages.

2) I like the idea of filtering out from the ConfigData object instead of adding another field.  Though I'm concerned about the performance hit requests might take to check every field in the configdata for type assertion.  I'll look into it and see if Go can do this gracefully.

3) Nesting is the original issue with the protobuf.  map[string]string only supports one layer of key/values.  I think it would be rare that someone would duplicate other config keys inside the ConfigData field.  We could add additional filtering to prevent APIID or OrgID (The 2 defaults inserted into Spec) but then that would add a precedent to filter for any other future fields.


I will now work on the suggested approach of using ConfigData and only applying first-level objects.